xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-send/xwiki-platform-mail-send-default/src/test/java/org/xwiki/mail/script/ScriptMimeMessageTest.java

```diff
@@ -90,4 +93,34 @@ public void addHeader() throws Exception
 
         assertEquals(testHeader, this.scriptMessage.getHeader("X-Test", null));
     }
+    
+    @Test
+    public void addPart() throws Exception
+    {
+        String header = "bodyPartTest";
+        MimeBodyPart bp = new MimeBodyPart();
+        bp.addHeader("BodyPart-Test", header);
+        this.scriptMessage.addPart(bp);
+        assertEquals(header, this.scriptMessage.getHeader("BodyPart-Test", null));
+    }
+    
+    @Test
+    public void addPartTwo() throws Exception
```

Comment:
Please rename the test method to something more meaningful, explaining what you're testing. For example: "addPartWhenNullMimeType". However, IMO we shouldn't support this, see above. I much prefer to throw an exception if the mimetype is null.